### PR TITLE
music_note.c, shape_spell.c, stat_change.c, tattle_window.c, recover.c data migration

### PR DESCRIPTION
**Pull request by JaThePlayer**
Matches many effect functions and migrates some data.

### CODE REVIEW

File: include/effects_internal.h [hunk 1]
```diff
@@ -66,7 +66,7 @@ FX_MAIN(stars_shimmer_main);
 FX_MAIN(rising_bubble_main);
 FX_MAIN(ring_blast_main);
 FX_MAIN(shockwave_main);
-FX_MAIN(music_note_main);
+void music_note_main(s32 arg0, f32 arg1, f32 arg2, f32 arg3);//
```
Comment (ethteck) [inline include/effects_internal.h:69]:
```suggestion
void music_note_main(s32, f32, f32, f32);
```

File: include/effects_internal.h [hunk 2]
```diff
@@ -134,7 +134,8 @@ FX_MAIN(whirlwind_main);
 FX_MAIN(red_impact_main);
 FX_MAIN(floating_cloud_puff_main);
 FX_MAIN(energy_in_out_main);
-FX_MAIN(tattle_window_main);
+//FX_MAIN(tattle_window_main);
+EffectInstance* tattle_window_main(s32 arg0, f32 arg1, f32 arg2, f32 arg3, f32 arg4, s32 arg5);
```
Comment (ethteck) [inline include/effects_internal.h:138]:
```suggestion
EffectInstance* tattle_window_main(s32, f32, f32, f32, f32, s32);
```

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 13 functions (+0.09%) on `us`.

File: src/effects/music_note.c [hunk 3]
```diff
@@ -2,13 +2,110 @@
...
+            break;
+    }
+}
+
+void music_note_init(EffectInstance* effect) {
 }
 
-INCLUDE_ASM(s32, "effects/music_note", music_note_update);
+void music_note_update(EffectInstance* effect) {
+    MusicNoteFXData* part;
```
Comment (ethteck) [inline src/effects/music_note.c:86]: MusicNoteFXData* part = ... (initialize on the same line)

File: include/effects_internal.h (hunk 1 above)
Comment (nanaian) [inline include/effects_internal.h:69]:
```suggestion
void music_note_main(s32 arg0, f32 arg1, f32 arg2, f32 arg3);
```

File: include/effects_internal.h [hunk 4]
```diff
@@ -134,7 +134,8 @@ FX_MAIN(whirlwind_main);
 FX_MAIN(red_impact_main);
 FX_MAIN(floating_cloud_puff_main);
 FX_MAIN(energy_in_out_main);
-FX_MAIN(tattle_window_main);
+//FX_MAIN(tattle_window_main);
```
Comment (nanaian) [inline include/effects_internal.h:137]:
```suggestion
```

File: src/effects/music_note.c [hunk 5]
```diff
@@ -2,13 +2,110 @@
 #include "effects_internal.h"
 
 void music_note_appendGfx(void* effect);
+void music_note_init(EffectInstance* effect);
+void music_note_update(EffectInstance* effect);
+void music_note_render(EffectInstance* effect);
 
-INCLUDE_ASM(s32, "effects/music_note", music_note_main);
+s32 D_E004C660[] = { 0x09001038, 0x090010C0, 0x09001148, 0x090011D0, 0x09001258, 0x090012E0, 0x09001368 };
```
Comment (ethteck) [inline src/effects/music_note.c:9]: These are display list pointers. can you do the thing we do in big_smoke_puff, for example?

```
extern Gfx D_09000FA0[];
extern Gfx D_09001060[];
extern Gfx D_09001120[];
extern Gfx D_090011E0[];
extern Gfx D_090012A0[];
extern Gfx D_09001360[];
extern Gfx D_09001420[];

Gfx* asdf[7] = {
    D_09000FA0, D_09001060, D_09001120, D_090011E0, D_090012A0, D_09001360, D_09001420
};
```
and then add the 0900 syms to undefined_syms.txt

File: src/effects/shape_spell.c [hunk 6]
```diff
@@ -23,8 +127,17 @@ void shape_spell_render(EffectInstance* effect) {
     retTask->renderMode |= RENDER_MODE_2;
 }
 
-INCLUDE_ASM(s32, "effects/shape_spell", func_E0024324);
+s32 func_E0024324(s32 arg0, s32 arg1) {
+    s32 frameCounter = gGameStatusPtr->frameCounter << 5;
```
Comment (nanaian) [inline src/effects/shape_spell.c:131]: Division?

**Review by ethteck — CHANGES_REQUESTED** [review]
A couple things to start, but I'll do a final review a bit later

File: src/effects/stat_change.c [hunk 7]
```diff
@@ -1,28 +1,230 @@
...
 
-void stat_change_render(void) {
+    part->unk_3C = 0;
+    part->unk_40 = 0;
+
+    do {} while (0); // required to match
+
+    part->pos.x += part->unk_30;
+    part->pos.y += part->unk_38;
+    part->pos.z += part->unk_34;
```
Comment (nanaian) [inline src/effects/stat_change.c:220]: You can probably make unk_30 a Vec3f

File: src/effects/tattle_window.c [hunk 8]
```diff
@@ -1,17 +1,154 @@
...
+    { .unk_00 = 0x0000, .unk_02 = 0xE6, .unk_03 = 0xFA },
+    { .unk_00 = 0x0000, .unk_02 = 0x78, .unk_03 = 0xF5 },
+    { .unk_00 = 0x0000, .unk_02 = 0x00, .unk_03 = 0xF0 },
+    { .unk_00 = 0x0000, .unk_02 = 0x00, .unk_03 = 0xD2 },
+    { .unk_00 = 0x0000, .unk_02 = 0x00, .unk_03 = 0xAA },
+    { .unk_00 = 0x0000, .unk_02 = 0x00, .unk_03 = 0x78 },
+    { .unk_00 = 0x0000, .unk_02 = 0x00, .unk_03 = 0x3C },
+};
+
+EffectInstance* tattle_window_main(s32 arg0, f32 arg1, f32 arg2, f32 arg3, f32 arg4, s32 arg5) {
```
Comment (nanaian) [inline src/effects/tattle_window.c:59]: `f32 arg1, f32 arg2, f32 arg3` are x y z

**Review by nanaian — CHANGES_REQUESTED** [review]

File: src/effects/stat_change.c (hunk 7 above)
Comment (JaThePlayer) [reply src/effects/stat_change.c:220]: well it would be weird that they do `pos.y += unk_30.z` and `pos.z += part->unk_34.y`

Comment (nanaian) [reply src/effects/stat_change.c:220]: oh, my bad lol

**Review by nanaian — APPROVED** [review]

File: src/effects/recover.c [hunk 9]
```diff
@@ -1,6 +1,14 @@
 #include "common.h"
 #include "effects_internal.h"
 
+void func_E0080448(EffectInstance* effect);
+
+s32 D_E0080AC0[] = { 0x09003540, 0x09003580, 0x090035C0, 0x09003600, 0x09003640, 0x09003680, 0x090036C0, 0x09003700, 0x09003740, 0x09003780, 0x090037C0, 0x09003800 };
```
Comment (ethteck) [inline src/effects/recover.c:6]: Same thing as earlier with the Gfxs

File: src/effects/recover.c [hunk 10]
```diff
@@ -1,6 +1,14 @@
...
+
+s32 D_E0080AF0[] = { 0x3E800000, 0x3F000000, 0x3F400000, 0x3F800000, 0x3F666666, 0x3F8CCCCD, 0x3F800000 };
```
Comment (ethteck) [inline src/effects/recover.c:8]: These are floats, so it's an f32 array just like the below one...weird that splat missed these. https://www.h-schmidt.net/FloatConverter/IEEE754.html

File: src/effects/shape_spell.c [hunk 11]
```diff
@@ -23,8 +127,17 @@ void shape_spell_render(EffectInstance* effect) {
...
-INCLUDE_ASM(s32, "effects/shape_spell", func_E0024324);
+s32 func_E0024324(s32 arg0, s32 arg1) {
+    s32 frameCounter = gGameStatusPtr->frameCounter * 32;
+    
+    return (f32)((shim_sin_deg(frameCounter + arg1) * (255 - arg0) + (255 - arg0)) * 0.5 + arg0);
+}
+
+s32 func_E00243BC(s32 arg0, s32 arg1) {
+    s32 frameCounter = gGameStatusPtr->frameCounter * 32;
+    arg1 += 0xB4;
```
Comment (ethteck) [inline src/effects/shape_spell.c:138]: empty line after decls, B4 in decimal please

File: src/effects/stat_change.c [hunk 12]
```diff
@@ -1,28 +1,230 @@
...
+    /* 0x28 */ f32 scaleX;
+    /* 0x2C */ f32 scaleY;
+    /* 0x30 */ f32 unk_30;
+    /* 0x34 */ f32 unk_34;
+    /* 0x38 */ f32 unk_38;
+    /* 0x3C */ s32 unk_3C;
+    /* 0x40 */ s32 unk_40;
+} StatChangeFXData;
+
+typedef struct extraArrowDataEntry {
```
Comment (ethteck) [inline src/effects/stat_change.c:28]: capital E to start the struct name

File: src/effects/stat_change.c [hunk 13]
```diff
@@ -1,28 +1,230 @@
...
+    return effect;
 }
 
-INCLUDE_ASM(s32, "effects/stat_change", stat_change_update);
+void stat_change_init(EffectInstance* effect) {
+}
+
+void stat_change_update(EffectInstance* effect) {
+    s32 old_unk18;
+    StatChangeFXData* part = effect->data;
```
Comment (ethteck) [inline src/effects/stat_change.c:181]: I'd put this line first and have the old_unk18 after

**Review by ethteck — CHANGES_REQUESTED** [review]
Found a few more things. Awesome job though! :) glad to have another on board for effect decomp. next I'll show you how to do graphics 😈

**Pull request merged**